packages/atlas/src/providers/notifications/notifications.hooks.ts

```diff
@@ -82,15 +87,21 @@ export const useNotifications = (opts?: Pick<QueryHookOptions, 'notifyOnNetworkS
   const [optimisticRead, setOptimisticRead] = useState<string[]>([])
   const notifications: NotificationRecord[] = useMemo(
     () =>
-      rawNotifications.map(
-        ({ node: { id, createdAt, status, notificationType } }): NotificationRecord => ({
+      rawNotifications.map(({ node: { id, createdAt, status, notificationType } }): NotificationRecord => {
+        const notificationData = parseNotificationType(notificationType as NotificationType)
+        if (notificationData.refetchAction && !refetchedNotifications.has(id)) {
+          const refetchFn = networkUtils[notificationData.refetchAction.name]
+          refetchedNotifications.set(id, 1)
+          refetchFn(...(notificationData.refetchAction.args as Parameters<typeof refetchFn>))
+        }
```

Comment:
LGTM!

Personally I would just move this if in a useEffect (with `[notifications, networkUtils]` as the dependencies).